src/aviary/dataset_server.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+import logging
+import time
+import traceback
+import uuid
+from contextlib import contextmanager
+from itertools import starmap
+from typing import Generic, TypeVar
+
+import uvicorn
+from pydantic import BaseModel, Field
+
+from aviary.env import Environment, TaskDataset
+from aviary.tools import MessagesAdapter, ToolRequestMessage, ToolsAdapter
+
+try:
+    from fastapi import FastAPI, HTTPException
+except ImportError:
+    # We will raise if a TaskDatasetServer is instantiated but fastapi is not available
+    FastAPI = HTTPException = None
+
+logger = logging.getLogger(__name__)
+
+
+class StartRequest(BaseModel):
+    task: int | None = None
+
+
+class EnvRequest(BaseModel):
+    env_id: str = Field(description="Maps to a running env ID")
```

Comment:
Can we have this be `str | UUID` for the ID?

Reply:
`env_id` is set by the server (to be a str), so there's no way for the user to force a UUID. Also supporting UUIDs requires additional (de)serialization code since pydantic doesn't natively support, and I'm not sure what benefit that would give us.